### ISSUE TITLE
Patch error on large composer update

### ISSUE THREAD

**Issue by DanieliMi** [vaimo/composer-patches#28]
Sometimes on large scale composer updates I get following error when the patches are supposed to be applied:

```Fatal error: Uncaught ArgumentCountError: Too few arguments to function Vaimo\ComposerPatches\Package\InfoResolver::__construct(), 1 passed in /builds/.../vendor/vaimo/composer-patches/src/Factories/PatchesLoaderFactory.php on line 48 and exactly 2 expected in /builds/.../vendor/vaimo/composer-patches/src/Package/InfoResolver.php:36
Fatal error: Uncaught ArgumentCountError: Too few arguments to function Vaimo\ComposerPatches\Package\InfoResolver::__construct(), 1 passed in /builds/.../vendor/vaimo/composer-patches/src/Factories/PatchesLoaderFactory.php on line 48 and exactly 2 expected in /builds/.../vendor/vaimo/composer-patches/src/Package/InfoResolver.php:36
Stack trace:
#0 /builds/.../vendor/vaimo/composer-patches/src/Factories/PatchesLoaderFactory.php(48): Vaimo\ComposerPatches\Package\InfoResolver->__construct(Object(Composer\Installer\InstallationManager))
#1 /builds/.../vendor/vaimo/composer-patches/src/Bootstrap.php(122): Vaimo\ComposerPatches\Factories\PatchesLoaderFactory->create(Object(Vaimo\ComposerPatches\Patch\DefinitionList\Loader\ComponentPool), Object(Vaimo\ComposerPatches\Config), true)
#2 /builds/.../vendor/vaimo/composer-patches/src/Bootstrap.php(102): Vaimo\ComposerPatches\Boo in /builds/.../vendor/vaimo/composer-patches/src/Package/InfoResolver.php on line 36
PHP Fatal error:  Uncaught ArgumentCountError: Too few arguments to function Vaimo\ComposerPatches\Package\InfoResolver::__construct(), 1 passed in /builds/.../vendor/vaimo/composer-patches/src/Factories/PatchesLoaderFactory.php on line 48 and exactly 2 expected in /builds/.../vendor/vaimo/composer-patches/src/Package/InfoResolver.php:36
Stack trace:
#0 /builds/.../vendor/vaimo/composer-patches/src/Factories/PatchesLoaderFactory.php(48): Vaimo\ComposerPatches\Package\InfoResolver->__construct(Object(Composer\Installer\InstallationManager))
#1 /builds/.../vendor/vaimo/composer-patches/src/Bootstrap.php(122): Vaimo\ComposerPatches\Factories\PatchesLoaderFactory->create(Object(Vaimo\ComposerPatches\Patch\DefinitionList\Loader\ComponentPool), Object(Vaimo\ComposerPatches\Config), true)
#2 /builds/.../vendor/vaimo/composer-patches/src/Bootstrap.php(102): Vaimo\ComposerPatches\Boo in /builds/.../vendor/vaimo/composer-patches/src/Package/InfoResolver.php on line 36`

Did anyone see this before? I am not quite sure how to reproduce it.

**Comment by allanpaiste** [comment]
This happens when updating the plugin where some classes or certain dependency setup might have changed.

Overall the plugin itself should always be updated with --no-plugins to make sure that it's not loaded nor triggered when it's files are replaced. 

The issue otherwise comes from the fact that half of the plugin is already loaded to memory when installation completes (that's the point where patches should apply).

**Comment by amenk** [comment]
@DanieliMi How can we put this in the CI pipleline?

something like `composer install vaimo/* ..no-plugins  && composer install` ?

**Comment by DanieliMi** [comment]
Thank you for the quick answer it makes sense.
So I guess we'll have to update this without plugins first and then update everything else.

@amenk I guess so :thinking:

**Comment by allanpaiste** [comment]
Idea: am now thinking that it might be wise to make sure on plugin run that every class has been loaded to memory ASAP (or at least to make sure that every important part has been instantiated) so that this would never happen.

**Comment by allanpaiste** [comment]
Addressed in 4.16.0 & 3.47.0

**Comment by PascalBrouwers** [comment]
Had this with `4.20.2` upgrading to `4.22.4`. Had to use `--no-plugins`
